aries-site/src/examples/templates/dashboards/content/chartCards/CostByMonth.js

```diff
@@ -0,0 +1,221 @@
+import { useContext, useEffect, useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Grid, ThemeContext, DataChart, Text } from 'grommet';
+import { ChartCard, Measure } from '../../components';
+import {
+  convertDatesToFeatures,
+  defaultWindow,
+  formatCurrency,
+  linearRegression,
+  mean,
+  REPORT_WINDOW_MAP,
+} from './utils';
+
+const MOCK_DATA = require('../../../../../data/mockData/consumption.json');
+
+export const CostByMonth = ({ period }) => {
+  const [values, setValues] = useState(null);
+  const [meanCost, setMeanCost] = useState(null);
+  const [projectedCost, setProjectedCost] = useState(null);
+  const [reportWindow, setReportWindow] = useState(defaultWindow);
+
+  const consumptionData = MOCK_DATA.consumption;
+  const theme = useContext(ThemeContext);
+  const chartColors = useMemo(
+    () =>
+      Object.entries(theme.global.colors)
+        .filter(([key]) => key.includes('graph'))
+        .map(([, value]) => value),
+    [theme.global.colors],
+  );
+
+  // Set reporting window
+  useEffect(() => {
+    setReportWindow(prevWindow => ({
+      ...prevWindow,
+      begin: new Date(
+        new Date().setDate(
+          prevWindow.end.getDate() - REPORT_WINDOW_MAP[period],
+        ),
+      ),
+    }));
+  }, [period]);
+
+  // Filter and assemble data set
+  useEffect(() => {
+    if (consumptionData) {
+      const nextValues = [];
+      consumptionData.forEach(datum => {
+        const dataPointDate = new Date(datum.endDate);
+        if (
+          dataPointDate > reportWindow.begin &&
+          dataPointDate < reportWindow.end
+        ) {
+          const key = datum.endDate;
+          const value = parseFloat(datum.cost.replace(/[$,]/gm, ''));
+          const index = nextValues.findIndex(el => el.key === key);
+
+          if (index === -1) {
+            nextValues.push({
+              key,
+              value,
+              color: undefined,
+            });
+          } else {
+            const nextValue = (
+              parseFloat(nextValues[index].value) + parseFloat(value)
+            ).toFixed(2);
+            nextValues[index].value = Number(nextValue);
+          }
+        }
+      });
+      nextValues
+        .sort((a, b) => b.key - a.key)
+        .forEach((value, index) => {
+          // eslint-disable-next-line no-param-reassign
+          value.color = chartColors[index % chartColors.length];
```

Comment:
Do we have to explicitly define the color? I thought that the data chart knew how to map through the color values

Reply:
also, I'm interpreting this as that it should result in different colors, but I'm only seeing one color in the graph. Or am I reading the % wrong ? :/

Reply:
We do not have to explicitly define color. I had left that in porting some of the logic over from the Pie Chart/Meter in case I needed it, but it can/should be removed.

In DataChart, colors map to each series presented. In this case it is only 'cost', so only one color. 

I could see "average cost" as another valuable series on this chart presented as a line, so that I'd have easy ability to see which months are over/under my average. If we were to add that, DataChart would automatically assign `graph-2` as the color, or we'd be able to overwrite it via an object on the `chart` prop.

Another valuable enhancement would be to plot the projected month and cost as an additional series. It too would receive a new chart color.

Reply:
Gotcha -- I could see "projected month" as an additional series being valuable -- that's common in graphs like this i feel.

Reply:
Removed.

I can add the projected month.